src/Pim/Bundle/CatalogBundle/ElasticSearch/Filter/AbstractFilter.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace Pim\Bundle\CatalogBundle\ElasticSearch\Filter;
+
+use Akeneo\Component\StorageUtils\Exception\InvalidPropertyException;
+use Pim\Bundle\CatalogBundle\ElasticSearch\SearchQueryBuilder;
+use Pim\Component\Catalog\Model\AttributeInterface;
+use Pim\Component\Catalog\Query\Filter\FilterInterface;
+use Pim\Component\Catalog\Validator\AttributeValidatorHelper;
+
+abstract class AbstractFilter implements FilterInterface
+{
+    /** @var SearchQueryBuilder */
+    protected $searchQueryBuilder = null;
+
+    /** @var array */
+    protected $supportedOperators = [];
+
+    /** @var AttributeValidatorHelper */
+    protected $attrValidatorHelper;
+
+    /** @var string[] */
+    protected $supportedAttributeTypes;
+
+    /** @var string[] */
+    protected $supportedFields;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsOperator($operator)
+    {
+        return in_array($operator, $this->supportedOperators);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getOperators()
+    {
+        return $this->supportedOperators;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getAttributeTypes()
+    {
+        return $this->supportedAttributeTypes;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsAttribute(AttributeInterface $attribute)
+    {
+        return in_array($attribute->getAttributeType(), $this->supportedAttributeTypes);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supportsField($field)
+    {
+        return in_array($field, $this->supportedFields);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getFields()
+    {
+        return $this->supportedFields;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setQueryBuilder($searchQueryBuilder)
+    {
+        if (!$searchQueryBuilder instanceof SearchQueryBuilder) {
+            throw new \InvalidArgumentException(
+                'Query builder should be an instance of "\Pim\Bundle\CatalogBundle\ElasticSearch\SearchQueryBuilder"'
```

Comment:
maybe a `sprintf` with `SearchQueryBuilder::class`, so if we move the class in the future, we don't have to update the message.